### PR TITLE
Add appveyor CI support

### PR DESCRIPTION
**Pull request by ShineQi**
Appveyor CI build configuration using mingw64/msys2. I've tested it on my branch, and the latest build log is here https://ci.appveyor.com/project/cwyyprog/bitshares-fc . The api.exe takes more time than Appveyor's limit, so is not in the build targets.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -133,7 +133,7 @@ else ( MSVC )
 endif ( MSVC )
 # End configure editline
 
-IF( WIN32 )
+IF( MSVC )
```
Comment (pmconrad) [inline CMakeLists.txt]: #121 replaces some `MSVC` with `WIN32`. Which one is correct / better? @cwyyprog @crypto-ape

File: appveyor.bash [hunk 2]
```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/bash
+export MSYSTEM=MINGW64
+wget http://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-boost-1.69.0-2-any.pkg.tar.xz
+pacman --noconfirm -U mingw-w64-x86_64-boost-1.69.0-2-any.pkg.tar.xz
+#pacman --noconfirm -S  mingw-w64-x86_64-boost; #mingw-w64-x86_64-openssl
+mkdir /msys64
+mount C:/msys64 /msys64;
+cd $APPVEYOR_BUILD_FOLDER/bitshares-fc;
+cmake -G "MSYS Makefiles" . -DOPENSSL_CONF_SOURCE="C:/msys64/mingw64/ssl/openssl.cnf" -DOPENSSL_ROOT_DIR=/mingw64 -DBOOST_ROOT=/mingw64;
+#cmake --build . -j 2 --target all_tests;
```
Comment (pmconrad) [inline appveyor.bash]: This one is important and should be built.

File: appveyor.bash [hunk 3]
```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/bash
+export MSYSTEM=MINGW64
```
Comment (pmconrad) [inline appveyor.bash:3]: Please add a `set -e` line to make the build fail if one of the steps fails.

File: appveyor.bash [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+cmake --build . -j 2 --target log_test;
+cmake --build . -j 2 --target bloom_test;
+cmake --build . -j 2 --target hmac_test;
+cmake --build . -j 2 --target blind;
+cmake --build . -j 2 --target task_cancel_test;
+#cmake --build . -j 2 --target api;
+cmake --build . -j 2 --target real128_test;
+cmake --build . -j 2 --target ecc_test;
+cmake --build . -j 2 --target bip_lock;
+echo "Build finished!"
```
Comment (pmconrad) [inline appveyor.bash:15]: Please run the tests after building. See `.travis.yml`.

Comment (pmconrad) [reply appveyor.bash:15]: Btw, shell scripts do not need a semicolon after each command. Only if another command follows on the same line.

File: appveyor.yml [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+    cd %APPVEYOR_BUILD_FOLDER%
+    
+    c:\msys64\usr\bin\bash -lc "pacman --noconfirm  -Syu"
+    
+    c:\msys64\usr\bin\bash -lc "pacman --noconfirm  -Syu"
+
+    C:\msys64\usr\bin\bash -lc $APPVEYOR_BUILD_FOLDER/bitshares-fc/appveyor.bash
+test: off
+deploy: off
```
Comment (pmconrad) [inline appveyor.yml]: Please add a terminating newline.

**Review by pmconrad — CHANGES_REQUESTED** [review]
Looks like a start. But all the test executables fail during linking with `ld.exe: cannot find -lrt`.

`api_test` is about to be removed, so never mind that.

File: CMakeLists.txt (hunk 1 above)
Comment (crypto-ape) [reply CMakeLists.txt]: Depends on context,
we agree on this line.

See [CMake variables](https://gitlab.kitware.com/cmake/community/wikis/doc/cmake/Checking-Platform) for more info.

 - `MSVC` 
    - targets one specific compiler, the Microsoft's Visual C/C++ Compiler
 - `WIN32` 
    - targets whole platform,
    - from my experience this includes `MSVC`, `MINGW` and also cross-compilation on Linux with MINGW compiler

**Comment by pmconrad** [comment]
@cwyyprog are you still working on this?
We've made quite a few changes to CMakeLists.txt, including the windows build.

**Comment by ShineQi** [comment]
No,  I'm not working on this due to lack of time. 



On Sat, Nov 2, 2019 at 4:23 PM +0800, "Peter Conrad" <notifications@github.com> wrote:












@cwyyprog are you still working on this?

We've made quite a few changes to CMakeLists.txt, including the windows build.

**Pull request closed without merging**